src/main/java/org/folio/rest/impl/PutOrderLineByIdHelper.java

```diff
@@ -170,13 +171,18 @@ public boolean isUpdateSuccessful() {
       .thenCompose(v -> inventoryHelper.getOrCreateHoldingsRecord(compPOL))
       .thenCompose(holdingsId -> inventoryHelper.handleItemRecords(compPOL, holdingsId))
       .thenCompose(itemIds -> {
-        // Temporal check. The idea is to create piece records for successfully created items and then throw exception
-        if (itemIds.size() != expectedItemsQuantity) {
-          throw new IllegalStateException("Expected items quantity does not correspond to created items");
-        }
-        return completedFuture(itemIds);
-      })
-      .thenCompose(itemIds -> createPieces(compPOL, itemIds));
+        int itemsSize = itemIds.size();
+
+        // Create piece records for successfully created items
+        return createPieces(compPOL, itemIds)
+          .thenAccept(v -> {
+            if (itemsSize != expectedItemsQuantity) {
+              String message = String.format("The issue happened creating items for PO Line with '%s' id. Expected %d but %d created",
```

Comment:
This block can be covered by unit tests only in scope of [MODORDERS-154](https://issues.folio.org/browse/MODORDERS-154). At the moment all items being created for POL have absolutely the same content. So there is no way to emulate partial items creation. In scope of [MODORDERS-154](https://issues.folio.org/browse/MODORDERS-154) items for POL would have different holding id's so partial items failure will be emulated as well.

Reply:
If this is unreachable code, we should consider commenting it out with a TODO and uncommenting it as part of [MODORDERS-154](https://issues.folio.org/browse/MODORDERS-154).  This would satisfy the code coverage requirement w/o compromising functionality

Reply:
I must be missing something... if we're testing against mocks, why can't we emulate this?

Reply:
The new test added to cover this case.